service/worker/migration/activities.go

```diff
@@ -675,34 +661,53 @@ func (a *activities) VerifyReplicationTasks(ctx context.Context, request *verify
 	// The verification step is retried for every VerifyInterval to handle #1. Verification progress
 	// is recorded in activity heartbeat. The verification is considered of making progress if there was at least one new execution
 	// being verified. If no progress is made for long enough, then
-	//  - more than RetryableTimeout, the activity fails with retryable error and activity will restart. This gives us the chance
-	//    to identify case #2 and #3 by rerunning createReplicationTasks.
-	//  - more than NonRetryableTimeout, it means potentially we encountered #4. The activity returns
-	//    non-retryable error and force-replication workflow will restarted.
-	for {
+	//  - more than checkSkipThreshold, it checks if outstanding workflow execution can be skipped locally (#2 and #3)
+	//  - more than NonRetryableTimeout, it means potentially #4. The activity returns
+	//    non-retryable error and force-replication will fail.
 
+	checkSkipCount := 0
```

Comment:
nit: the name of this variable is a little bit hard to follow
If you cannot think of a better name, at least leave a comment explaining why you want to do this.

Reply:
code removed.